crates/wit-component/src/encoding.rs

```diff
@@ -2093,7 +2093,8 @@ impl<'a> ImportEncoder<'a> {
 #[derive(Default)]
 pub struct ComponentEncoder {
     module: Vec<u8>,
-    encoding: StringEncoding,
+    /// String encoding
+    pub encoding: StringEncoding,
```

Comment:
Instead of making this field public I think you should be able to use the `ComponentEncoder::encoding` method.